ckan/views/group.py

```diff
@@ -0,0 +1,1094 @@
+
+
+def _index_template(group_type):
+    return lookup_group_plugin(group_type).index_template()
+
+
+def _group_form(group_type=None):
+    return lookup_group_plugin(group_type).group_form()
+
+
+def _form_to_db_schema(group_type=None):
+    return lookup_group_plugin(group_type).form_to_db_schema()
+
+
+def _db_to_form_schema(group_type=None):
+    '''This is an interface to manipulate data from the database
+    into a format suitable for the form (optional)'''
+    return lookup_group_plugin(group_type).db_to_form_schema()
+
+
+def _setup_template_variables(context, data_dict, group_type=None):
+    return lookup_group_plugin(group_type).\
+        setup_template_variables(context, data_dict)
+
+
+def _new_template(group_type):
+    return lookup_group_plugin(group_type).new_template()
+
+
+def _about_template(group_type):
+    return lookup_group_plugin(group_type).about_template()
+
+
+def _read_template(group_type):
+    return lookup_group_plugin(group_type).read_template()
+
+
+def _history_template(group_type):
+    return lookup_group_plugin(group_type).history_template()
+
+
+def _edit_template(group_type):
+    return lookup_group_plugin(group_type).edit_template()
+
+
+def _activity_template(group_type):
+    return lookup_group_plugin(group_type).activity_template()
+
+
+def _admins_template(group_type):
+    return lookup_group_plugin(group_type).admins_template()
+
+
+def _bulk_process_template(group_type):
+    return lookup_group_plugin(group_type).bulk_process_template()
+# end hooks
+
+
+def _replace_group_org(string):
+    ''' substitute organization for group if this is an org'''
+    return string
+
+
+def _action(action_name):
+    ''' select the correct group/org action '''
+    return get_action(_replace_group_org(action_name))
+
+
+def _check_access(action_name, *args, **kw):
+    ''' select the correct group/org check_access '''
+    return check_access(_replace_group_org(action_name), *args, **kw)
+
+
+def _render_template(template_name, group_type):
+    ''' render the correct group/org template '''
+    return base.render(
+        _replace_group_org(template_name),
+        extra_vars={'group_type': group_type})
+
+
+def _force_reindex(grp):
+    ''' When the group name has changed, we need to force a reindex
+    of the datasets within the group, otherwise they will stop
+    appearing on the read page for the group (as they're connected via
+    the group name)'''
+    group = model.Group.get(grp['name'])
+    for dataset in group.packages():
+        search.rebuild(dataset.name)
+
+
+@group.before_request
+def before_request():
+    try:
+        context = dict(model=model, user=g.user, auth_user_obj=g.userobj)
+        check_access(u'site_read', context)
+    except NotAuthorized:
+        _, action = request.url_rule.endpoint.split(u'.')
+        if action not in (u'group_list', ):
+            base.abort(403, _(u'Not authorized to see this page'))
+
+
+def index():
+    group_type = request.blueprint
```

Comment:
This won't work for custom group types (ie if via `IGroupForm` we change `/group` to `/theme`. I think the approach we are currently following on master is safer, using `self._guess_group_type()`.

Of course this is to get the group `type` (ie `group`, `theme` , `collection` etc). If we need to known if the underlying class is a group or an org (eg `is_organization`) we could so something like:
```python
def index(is_organization):
     pass

group.add_url_rule(u'/', view_func=index, strict_slashes=False, defaults={'is_organization': False})

organization.add_url_rule(u'/', view_func=index, strict_slashes=False, defaults={'is_organization': True})


```

And the same when we register the custom routes on `IGroupForm`

Reply:
But I don't think that would be necessary because the current controller already deals with this with the `_replace_group_org()` methods. So let's try with just `self._guess_group_type()`